plugins/module_utils/network/common/network.py

```diff
@@ -202,8 +202,8 @@ def get_resource_connection(module):
     if network_api == "netconf":
         module._connection = NetconfConnection(module._socket_path)
     elif network_api == "local":
-        # This isn't supported, but we shouldn't fail here.
-        # Set the connection to a fake connection so it fails sensibly.
+        # not accessible code alert can be taken out at around  01-01-2027,
+        # when connection local is removed
         module._connection = LocalResourceConnection(module)
```

Comment:
This is exclusively for resource modules which have _never_ supported local connection. This check was added defensively in case a facts module, over a local connection, tried to query for resource module facts. It couldn't _fail_ here... that would kill the entire facts module without a useful way to figure out what happened, so the fake connection is returned so that it can fail gracefully when querying the resource facts and still return legacy facts when running on `connection: local`.
ref. https://github.com/ansible/ansible/pull/64570

Needless to say, none of that is needed anymore. This branch and the `LocalResourceConnection` below can both be removed, and the AssertionError on lines 222-223 can be upgraded into a failure, as no collection with resource modules should consider local a valid connection.